tests/unit/App.spec.js

```diff
@@ -89,4 +89,55 @@ Given("an App", () => {
       expect(app.vm.url).to.equal(expectedUrl);
     });
   });
+
+  When("a response is returned", () => {
+    beforeEach(async () => {
+      app.setData({
+        wasResponseAnError: false,
+        statusText: ""
+      });
+
+      app.setMethods({
+        request: app.vm.handleResponse
+      });
+    });
+
+    And("the response was a 4xx", () => {
+      const response = {
+        status: 404,
+        statusText: "Not Found"
+      };
+
+      beforeEach(async () => {
+        app.vm.request(response);
+      });
+
+      Then("the response was an error", () => {
+        expect(app.vm.wasResponseAnError).to.equal(true);
```

Comment:
you should be testing that Code's "wasResponseAnError" prop is equal to true, not app's"wasResponseAnError". You want to be testing the behavior, not implementation. https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

Reply:
Done